File: app/src/main/java/nyc/c4q/hakeemsackes_bramble/infinity_loop_clone/tileRecyclerView/TileViewHolder.java
package nyc.c4q.hakeemsackes_bramble.infinity_loop_clone.tileRecyclerView;

import android.support.v7.widget.RecyclerView;
import android.view.View;

import nyc.c4q.hakeemsackes_bramble.infinity_loop_clone.custom_views.TileView;
import nyc.c4q.hakeemsackes_bramble.infinity_loop_clone.gameLevelObjects.GameOneLayout;
import nyc.c4q.hakeemsackes_bramble.infinity_loop_clone.gameLevelObjects.Tile;

/**
 * Created by hakeemsackes-bramble on 10/15/17.
 */

class TileViewHolder extends RecyclerView.ViewHolder {
    private GameOneLayout mGameOneLayout;
    private static final String TAG = TileViewHolder.class.getName();

    public TileViewHolder(View itemView, GameOneLayout gameOneLayout) {
        super(itemView);
        this.mGameOneLayout = gameOneLayout;
    }

    public void bind(final int position) {

        final Tile tile = mGameOneLayout.getGameTiles().get(position);
        ((TileView) itemView).implementTileValues(tile);
        mGameOneLayout.runCheckTileAlignmentListener(tile, position);
        mGameOneLayout.addCorrectedTile(position, tile.isProperlyAligned());

        itemView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                /**
                 * here i'll run color change animations
                 */
                if (!mGameOneLayout.hasAllTilesAligned()) {
                    ((TileView) itemView).rotateView(tile.getOrientation(), (tile.getOrientation() + 1) % 4);
                    tile.assignOrientation((tile.getOrientation() + 1) % 4);
                    mGameOneLayout.runCheckTileAlignmentListener(tile, position);
                }
            }
        });


    }
}

File: app/src/main/java/nyc/c4q/hakeemsackes_bramble/infinity_loop_clone/tileRecyclerView/TileViewHolder.kt
package nyc.c4q.hakeemsackes_bramble.infinity_loop_clone.tileRecyclerView

import android.support.v7.widget.RecyclerView.ViewHolder
import android.view.View
import nyc.c4q.hakeemsackes_bramble.infinity_loop_clone.custom_views.TileView
import nyc.c4q.hakeemsackes_bramble.infinity_loop_clone.gameLevelObjects.GameLayout

/**
 * Created by hakeemsackes-bramble on 10/15/17.
 */
class TileViewHolder(itemView: View?, private val mGameLayout: GameLayout?) : ViewHolder(
    itemView!!
) {
    fun bind(position: Int) {
        val tile = mGameLayout.getGameTiles()[position]
        (itemView as TileView).setTileValues(tile)
        mGameLayout!!.runCheckTileAlignmentListener(tile, position)
        mGameLayout.addCorrectedTile(position, tile!!.isProperlyAligned)
        itemView.setOnClickListener {
            /**
             * here i'll run color change animations
             */
            /**
             * here i'll run color change animations
             */
            if (!mGameLayout.hasAllTilesAligned()) {
                (itemView as TileView).rotateView(tile!!.orientation, (tile!!.orientation + 1) % 4)
                tile!!.orientation = (tile!!.orientation + 1) % 4
                mGameLayout.runCheckTileAlignmentListener(tile, position)
            }
        }
    }

    companion object {
        private val TAG = TileViewHolder::class.java.name
    }
}